src/solver/algorithms/Fallin_solver.py

```diff
@@ -0,0 +1,223 @@
+# Fallin_solver.py
+# author: Lenny Flores
+# date: 4-24-2015
+
+class FallinSolver(object):
+    """Initialize cells and count matrices"""
+    def __init__(self):    
+        self.cells = []
+        self.count = []
+        for i in range(9):
+            self.cells += [ [] ]
+            self.count += [ [] ]
+            for j in range(9):
+                self.cells[i] += [0]
+                self.count[i] += [0]
+        self.known = 0
+    
+    def copy(self):
+        """Makes a copy of the sudoku data
+        Instantiate the FallingSolver class and then, makes a copy of game at the current state and assign them to
+        the attributes of class, then return the array of numbers that represent the game solved at the current state
+        """
+        copy = FallinSolver()
+        for i in range(9):
+            for j in range(9):
+                copy.cells[i][j] = self.cells[i][j]
+                copy.count[i][j] = self.count[i][j]
+                copy.known = self.known
+        return copy
+
+    def load(self, data):
+        """Loads from external data
+        Data is a simple array of array of matrix of numbers that represent the sudoku game pending to solve"""
+        self.known = 0
+        for i in range(9):
+            for j in range(9):
+                self.compare_cell(i, j, data)
+                
+    def dump(self):
+        """Dumps to external data (ie, a simple matrix/array-of-arrays)
+        dump returns the sudoku game solution on a matrix output type
+        """
+        out = []
+        for i in range(9):
+            out += [ [] ]
+            for j in range(9):
+                self.out_cell(i, j, out)        
+        return out
+        
+    def load_string(self, str):
+        """Transform string to a set of row and lines, without the spaces and line breaks str, receive the sudoku game as an string with format and then values 
+        are stored in lines and rows lists respectively
+        """
+        block_of_nine = str.split("\n");
+        lines = []
+        self.verify_block_of_nine(block_of_nine, lines)        
+    
+    def verify_block_of_nine(self, block_of_nine, lines):
+        """verify in a block of nine"""
+        for line in block_of_nine:
+            line = line.strip()            
+            if(line == ''):
+                continue
+            row = []            
+            block_of_three = line.split()
+            self.verify_block_of_three(block_of_three, line, row)          
+            lines += [row]
+        self.load(lines)
+    
+    def verify_block_of_three(self, block_of_three, line, row):
+        """Verify in a block of three"""
```

Comment:
no documentation for the parameters and the description of the method does not help at all to understand what does the method